nengo_bones/templates/setup.cfg.template

```diff
@@ -85,49 +83,22 @@ markers =
     {{ marker }}: {{ desc }}
     {% endfor %}
 {% endif %}
-{% if pytest.filterwarnings %}
-filterwarnings =
-    {% for filter in pytest.filterwarnings %}
-    {{ filter }}
+{% for param, (type, vals) in pytest.extras.items() %}
+  {% if type == "str" %}
+{{ param }} = {{ vals }}
+  {% elif type == "list" %}
+{{ param }} =
+    {% for val in vals %}
+    {{ val }}
     {% endfor %}
-{% endif %}
-{% if pytest.nengo_simulator %}
-nengo_simulator = {{ pytest.nengo_simulator }}
-{% endif %}
-{% if pytest.nengo_simloader %}
-nengo_simloader = {{ pytest.nengo_simloader }}
-{% endif %}
-{% if pytest.nengo_neurons %}
-nengo_neurons =
-{% for neurontype in pytest.nengo_neurons %}
-    {{ neurontype }}
-{% endfor %}
-{% endif %}
-{% if pytest.nengo_test_unsupported %}
-nengo_test_unsupported =
-{% for var, val in pytest.nengo_test_unsupported.items() %}
+  {% elif type == "dict" %}
+{{ param }} =
+    {% for var, val in vals.items() %}
     {{ var }}
         "{{ val | wordwrap(break_long_words=False, wrapstring="\n        ") }}"
+    {% endfor %}
+  {% endif %}
```

Comment:
This is probably a personal preference thing, but for me, I prefer if the indentation of the control structures (e.g. for/if) matches the indentation of the text being formatted.  So, e.g., I would format this like
``` python
{% for param ... %}
...
{% elif type ... %}
{{ param }} =
    {% for ... %}
    {{ var }}
        "{{ ...
     {% endfor %}
{% endif %}
{% endfor %}
```
You lose the scoping information in the control blocks, but I find it's easier to see what the resulting text would look like. Otherwise when you're scanning each row you have to mentally parse "is this a control-block indent? or a formatting indent?".

Reply:
I did it that way because I kept missing `endfor`s and `endif`s or having them in the wrong order, was originally going to flatten it all but forgot. In any case, now that it's working I'm fine with whatever indentation for the control blocks :+1: